java/java-common/src/main/java/org/cloudburstmc/protocol/java/packet/play/clientbound/LevelChunkPacket.java

```diff
@@ -0,0 +1,37 @@
+package org.cloudburstmc.protocol.java.packet.play.clientbound;
+
+import com.nukkitx.nbt.NbtMap;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.cloudburstmc.protocol.java.JavaPacket;
+import org.cloudburstmc.protocol.java.handler.JavaPlayPacketHandler;
+import org.cloudburstmc.protocol.java.packet.type.JavaPacketType;
+import org.cloudburstmc.protocol.java.packet.type.JavaPlayPacketType;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode(doNotUseGetters = true, callSuper = false)
+public class LevelChunkPacket extends JavaPacket<JavaPlayPacketHandler> {
+    private int x;
+    private int z;
+    private int availableSections;
+    private NbtMap heightmaps;
+    @Nullable
+    private int[] biomes;
+    private byte[] buffer;
```

Comment:
Should this be a ByteBuf? I know that DaPorkchop thinks that Bedrock's LevelChunkPacket byte array should be a bytebuffer.

Reply:
In bedrock part of proto v3 we will try to limit creating new allocations. I assume we will use netty ByteBuf or ByteBuffer based on source buffer in LevelChunk and similar packets.